script/configure-assets.js

```diff
@@ -1,66 +1,38 @@
-    const manifest = {};
-    Object.keys(originalManifest).forEach(originalManifestKey => {
-      const matchData = originalManifestKey.match(/(.*)\.js$/);
-      if (matchData !== null) {
-        const newKey = `${matchData[1]}.entry.js`;
-        manifest[newKey] = originalManifest[originalManifestKey];
-      } else {
-        manifest[originalManifestKey] = originalManifest[originalManifestKey];
-      }
-    });
-
-    // For each file in the build, if it is a HTML or CSS file, loop over all
-    // the keys in the manifest object and do a search and replace for the
-    // key with the value.
-    Object.keys(files).forEach(filename => {
-      if (filename.match(/\.(html|css)$/) !== null) {
-        Object.keys(manifest).forEach(originalAssetFilename => {
-          const newAssetFilename = manifest[originalAssetFilename].replace(
-            '/generated/',
-            '',
-          );
-          const file = files[filename];
-          const contents = file.contents.toString();
-          const regex = new RegExp(originalAssetFilename, 'g');
-          file.contents = new Buffer(contents.replace(regex, newAssetFilename));
-        });
-      }
-    });
-
-    // Create a copy of the proxy-write files without cache-bust hashes
-    [
-      'proxy-rewrite.entry.js',
-      'styleConsolidated.css',
-      'static-pages.css',
-      'vendor.entry.js',
-      'polyfills.entry.js',
-    ].forEach(unhashedName => {
-      const hashedName = manifest[unhashedName];
-
-      files[`generated/${unhashedName}`] = files[hashedName.substr(1)]; // eslint-disable-line no-param-reassign
-    });
-
-    done();
-  };
+/* eslint-disable no-param-reassign */
+
+const watch = require('metalsmith-watch');
+const environments = require('./constants/environments');
+const webpackMetalsmithConnect = require('../config/webpack-metalsmith-connect');
+const downloadAssets = require('./download-assets');
+const addAssetHashes = require('./add-asset-hashes');
+
+function configureAssets(smith, buildOptions) {
+  const isContentDeployment = buildOptions['content-deployment'];
+  const isDevBuild = [environments.DEVELOPMENT, environments.VAGOVDEV].includes(
+    buildOptions.buildtype,
+  );
+
+  if (buildOptions.watch) {
+    const watchPaths = {
+      [`${buildOptions.contentRoot}/**/*`]: '**/*.{md,html}',
+      [`${buildOptions.contentPagesRoot}/**/*`]: '**/*.{md,html}',
+    };
+
+    const watchMetalSmith = watch({ paths: watchPaths, livereload: true });
+
+    smith.use(watchMetalSmith);
+    smith.use(webpackMetalsmithConnect.watchAssets(buildOptions));
+  } else {
+    if (isContentDeployment) {
+      smith.use(downloadAssets(buildOptions));
+    } else {
+      smith.use(webpackMetalsmithConnect.compileAssets(buildOptions));
+    }
+
+    if (!isDevBuild) {
+      smith.use(addAssetHashes(buildOptions));
+    }
+  }
 }
 
-module.exports = addAssetHashes;
+module.exports = configureAssets;
```

Comment:
This logic was mostly just chunked from the `build.js`. However, it includes that `isContentDeployment` flag to determine whether to compile or to download assets.